com.musala.xml.parser.executable/src/com/musala/xml/parser/executable/Main.java

```diff
@@ -0,0 +1,257 @@
+package com.musala.xml.parser.executable;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.TreeMap;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Unmarshaller;
+
+import com.musala.xml.model.executable.School;
+import com.musala.xml.model.executable.Student;
+import com.musala.xml.model.executable.Teacher;
+
+public class Main {
+
+    public static void main(String[] args) throws IOException, JAXBException, Exception {
+
+        ClassLoader classload = Main.class.getClassLoader();
+        File configFile = null;
+        try {
+            configFile = new File(args[0]);
+        } catch (Exception ex) {
+            System.err.println(ex.getMessage());
+        }
+
+        JAXBContext jaxb = null;
+        Unmarshaller jaxbUnmarsheller = null;
+        try {
+            jaxb = JAXBContext.newInstance(School.class);
+            jaxbUnmarsheller = jaxb.createUnmarshaller();
+        } catch (JAXBException ex) {
+            System.err.println(ex.getMessage());
+        }
+
+        School mySchool = null;
+        try {
+            mySchool = (School) jaxbUnmarsheller.unmarshal(configFile);
+        } catch (NullPointerException ex) {
+            System.err.println("The school is empty");
+        }
+
+        TreeMap<String, String> keyValuePair = new TreeMap<>();
+        PrintWriter keyValueWriter = new PrintWriter(args[2]);
+        Print(mySchool, args[1], keyValuePair, keyValueWriter);
+    }
+
+    private static ClassNumberText turnClassNumberIntoText(int numberOfClass) {
+        ClassNumberText word;
+        switch (numberOfClass) {
+            case 0:
+                word = ClassNumberText.first;
+                break;
+            case 1:
+                word = ClassNumberText.second;
+                break;
+            case 2:
+                word = ClassNumberText.third;
+                break;
+            case 3:
+                word = ClassNumberText.fourth;
+                break;
+            case 4:
+                word = ClassNumberText.fifth;
+                break;
+            case 5:
+                word = ClassNumberText.sixth;
+                break;
+            case 6:
+                word = ClassNumberText.seventh;
+                break;
+            case 7:
+                word = ClassNumberText.eighth;
+                break;
+            case 8:
+                word = ClassNumberText.ninth;
+                break;
+            default:
+                word = ClassNumberText.nth;
+        }
+
+        return word;
+    }
+
+    private static String getTheObjectProp(Object currentObj, int numberOfProp, String nameOfObject)
+            throws NullPointerException {
+        switch (nameOfObject) {
+            case "Student":
```

Comment:
A single external method should be used in both Student and Teacher case, since 3/4 of the properties which are returned are the same for both teacher and student classes. You can declare the method to accept a boolean argument so that if a student object is requested, the additional faculty number class property is examined as well.